ultra/ultra/train.py

```diff
@@ -94,6 +147,53 @@ def train(
         for agent_id, agent_spec in agent_specs.items()
     }
 
+    # Define an 'etag' for this experiment's data directory based off policy_classes.
+    # E.g. From a ["ultra.baselines.dqn:dqn-v0", "ultra.baselines.ppo:ppo-v0"]
+    # policy_classes list, transform it to an etag of "dqn-v0:ppo-v0".
+    etag = ":".join([policy_class.split(":")[-1] for policy_class in policy_classes])
+
+    return agent_ids, agent_classes, agent_specs, agents, etag
+
+
+def _save_agent_metadata(experiment_dir, agent_ids, agent_classes, agent_specs):
+    # Save relevant agent metadata.
+    if not os.path.exists(f"{experiment_dir}/agent_metadata.pkl"):
+        if not os.path.exists(experiment_dir):
+            os.makedirs(experiment_dir)
+        with open(f"{experiment_dir}/agent_metadata.pkl", "wb") as metadata_file:
+            dill.dump(
+                {
+                    "agent_ids": agent_ids,
+                    "agent_classes": agent_classes,
+                    "agent_specs": agent_specs,
+                },
+                metadata_file,
+                pickle.HIGHEST_PROTOCOL,
+            )
```

Comment:
What are your thoughts on if we check if `agent_metadata.pkl` exists outside of the function? So `_save_agent_metadata` would look something like:
```suggestion
def _save_agent_metadata(experiment_dir, filename, agent_ids, agent_classes, agent_specs):
    # Save relevant agent metadata.
    if not os.path.exists(experiment_dir):
        os.makedirs(experiment_dir)
    with open(os.path.join(experiment_dir, filename), "wb") as metadata_file:
        dill.dump(
            {
                "agent_ids": agent_ids,
                "agent_classes": agent_classes,
                "agent_specs": agent_specs,
            },
            metadata_file,
            pickle.HIGHEST_PROTOCOL,
        )
```
Then we would have to check to see if the file exists before we call this function.

If we do it like this, the function may be more descriptive of its name, but let me know what you think.